docs/publish.py

```diff
@@ -1,41 +1,76 @@
 #!/usr/bin/python3
 import os
 import sys
-__dir__ = os.path.dirname(__file__)
-sys.path.insert(0, os.path.join(__dir__, ".."))
+import typing as ty
+
+script_dir = os.path.dirname(os.path.realpath(__file__))
+sys.path.insert(0, os.path.join(script_dir, ".."))
 
 import sphinx.cmd.build
 import ipfshttpclient
 
-# Ensure working directory is script directory
-os.chdir(__dir__)
+os.chdir(script_dir)
+
+
+def main(argv: ty.List[str]) -> int:
+	if len(argv) >= 1:
+		ipns_key = argv[0]
+	else:
+		ipns_key = None
 
-def main(argv=sys.argv[1:], program=sys.argv[0]):
-	if len(argv) != 1:
-		print("Usage: {0} [IPNS-key]".format(os.path.basename(program)))
+		print("Usage: {0} [IPNS-key]".format(os.path.basename(__file__)))
```

Comment:
There was a bug where the script could not be executed directly from its own folder and had to be referenced as `docs/publish.py` instead of `publish.py` - that is fixed in this PR.